pulpcore/app/tasks/purge.py

```diff
@@ -0,0 +1,55 @@
+from gettext import gettext as _
+from django_currentuser.middleware import get_current_authenticated_user
+from pulpcore.app.models import (
+    ProgressReport,
+    Task,
+)
+from pulpcore.app.role_util import get_objects_for_user
+
+
+def purge(finished_before, states):
+    """
+    This task purges from the database records of tasks which finished prior to the specified time.
+
+    It will remove only tasks that are 'owned' by the current-user (admin-users own All The Things,
+    so admins can delete all tasks).
+
+    It will not remove tasks that are incomplete (ie, in states running|waiting|cancelling).
+
+    It reports (using ProgressReport) the total entities deleted, as well as individual counts
+    for each class of entity. This shows the results of cascading-deletes that are triggered
+    by deleting a Task.
+
+    args:
```

Comment:
nitpick feel free to ignore
```suggestion
    Args:
```